src/repl-frontend/scala/tools/nsc/MainGenericRunner.scala

```diff
@@ -58,17 +58,17 @@ class MainGenericRunner {
       }
 
       import GenericRunnerCommand.{AsObject, AsScript, AsJar, Error}
-      def runTarget(): Either[Throwable, Boolean] = howToRun match {
+      def runTarget(): Option[Throwable] = howToRun match {
         case AsObject =>
           ObjectRunner.runAndCatch(settings.classpathURLs, thingToRun, command.arguments)
         case AsScript if isE =>
-          Right(ScriptRunner.runCommand(settings, combinedCode, thingToRun +: command.arguments))
+          ScriptRunner(settings).runScriptText(combinedCode, thingToRun +: command.arguments)
         case AsScript =>
-          ScriptRunner.runScriptAndCatch(settings, thingToRun, command.arguments)
+          ScriptRunner(settings).runScript(thingToRun, command.arguments)
         case AsJar    =>
           JarRunner.runJar(settings, thingToRun, command.arguments)
         case Error =>
-          Right(false)
+          None
```

Comment:
This changes the exit value of `scala Junk` from `1` to `0`. `howToRun` sees `no such class`.

Are there conditions where the other runners did not throw but would return `Some(false)` for failure? Check failed compilation, for example.